### ISSUE TITLE
provide solution to run in browser

### ISSUE THREAD

**Issue by cedric05** [cedric05/dothttp#39]
currently there is an option https://pyodide.org/ which coverts python module to webassembly, which works in browser. 
lets isolate request generation part and provide it as module.
browsers/other js runtimes can use this to run.

**Comment by cedric05** [comment]
here is poc. textx can be run in browser. all we need to do is to do is get object from dothttp. 

```html
<html>
<head>
  <meta charset="utf-8">
</head>
<body>
  <script type="text/javascript">
      // set the Pyodide files URL (packages.json, pyodide.asm.data etc)
      window.languagePluginUrl = 'https://cdn.jsdelivr.net/pyodide/v0.16.1/full/';
  </script>
  <script type="text/javascript" src="https://cdn.jsdelivr.net/pyodide/v0.16.1/full/pyodide.js"></script>
  <script type="text/javascript">
    pythonCode = `
      def do_work(*args):
        from textx import metamodel_from_str, get_children_of_type

        grammar = """
        Model: commands*=DrawCommand;
        DrawCommand: MoveCommand | ShapeCommand;
        ShapeCommand: LineTo | Circle;
        MoveCommand: MoveTo | MoveBy;
        MoveTo: 'move' 'to' position=Point;
        MoveBy: 'move' 'by' vector=Point;
        Circle: 'circle' radius=INT;
        LineTo: 'line' 'to' point=Point;
        Point: x=INT ',' y=INT;
        """
        
        class Point(object):
            def __init__(self, parent, x, y):
                self.parent = parent
                self.x = x
                self.y = y
        
            def __str__(self):
                return "{},{}".format(self.x, self.y)
        
            def __add__(self, other):
                return Point(self.parent, self.x + other.x, self.y + other.y)
        
        mm = metamodel_from_str(grammar, classes=[Point])
        
        model_str = """
            move to 5, 10
            line to 10, 10
            line to 20, 20
            move by 5, -7
            circle 10
            line to 10, 10
        """
        
        model = mm.model_from_str(model_str)
        
        
        def cname(o):
            return o.__class__.__name__
        
        position = Point(None, 0, 0)
        for command in model.commands:
            if cname(command) == 'MoveTo':
                print('Moving to position', command.position)
                position = command.position
            elif cname(command) == 'MoveBy':
                position = position + command.vector
                print('Moving by', command.vector, 'to a new position', position)
            elif cname(command) == 'Circle':
                print('Drawing circle at', position, 'with radius', command.radius)
            else:
                print('Drawing line from', position, 'to', command.point)
                position = command.point
        print('End position is', position)
        
        
        points = get_children_of_type("Point", model)
        for point in points:
            print('Point: {}'.format(point))
      import micropip
      micropip.install('textx').then(do_work)
    `

    languagePluginLoader.then(() => {
      return pyodide.loadPackage(['micropip'])
    }).then(() => {
      pyodide.runPython(pythonCode);
    })
  </script>
</body>
</html>
```

**Comment by cedric05** [comment]
- [ ] provide .wasm, .js  (with minimalist packages)
- [ ] parse and provide object to js runtime
- [ ] js runtime will unwrap object and uses it to run curl

**Comment by cedric05** [comment]
(wasm)[https://github.com/cedric05/dothttp/tree/wasm] branch has poc changes.

loadable html code is here.
```html
<html>
<head>
  <meta charset="utf-8">
</head>
<body>
  <script type="text/javascript">
      // set the Pyodide files URL (packages.json, pyodide.asm.data etc)
      window.languagePluginUrl = 'https://cdn.jsdelivr.net/pyodide/v0.16.1/full/';
  </script>
  <script type="text/javascript" src="https://cdn.jsdelivr.net/pyodide/v0.16.1/full/pyodide.js"></script>
  <script type="text/javascript">
    pythonCode = `
      def haha(*args, **kwargs):
        from dothttp import Config, HttpDefBase
        class content_override(HttpDefBase):
            def load_content(self):
                self.original_content = self.content = """@name(test)
                POST https://httpbin.org/post
                ? haha = ramu
                ? haha = ramu
                // ? haha2 = ramu
                ? haha3 = ramu
                json({
                    "this is first level" : "this is second level",
                    "this is first level2" : "this is second level"
                })
                """
        def main():
            out = content_override(Config(target="1", no_cookie=True, property_file=None, experimental=False, format=False,
                                        stdout=False, debug=True, info=False, curl=False, env=[], file="", properties=[]))
            out.load()
            out.load_def()
            print(out.httpdef)
        main()
      import micropip
      micropip.install(['textx', 'http://localhost:8000/dothttp_req-0.0.10.dev0-py3-none-any.whl']).then(haha)
      
    `
    languagePluginLoader.then(() => {
      return pyodide.loadPackage(['micropip'])
    }).then(() => {
      pyodide.runPython(pythonCode);
    })
  </script>
</body>
</html>
```

**Comment by cedric05** [comment]
packaging current solution is still pending, hopefully we will get some answers...
https://github.com/iodide-project/pyodide/discussions/1379

**Comment by cedric05** [comment]
code changes are available in v0.0.10, checkout 
https://pypi.org/project/dothttp-req-wasm/
